statsd/statsd.go

```diff
@@ -489,15 +497,29 @@ func (c *Client) SimpleServiceCheck(name string, status ServiceCheckStatus) erro
 	return c.ServiceCheck(sc)
 }
 
+// shutdown shuts down the goroutines associated with the client and waits for them.
+// It returns true if shutdown was performed, and false if the client was already shut down.
+func (c *Client) shutdown() bool {
```

Comment:
IMO this will be much easier to follow if we drop this method and just move this logic into `Close`. The bool return value is slightly unusual. I don't see why we'd need a separate method. I think a good reason to have a separate method can be:
* parent method is getting huge and code is hard to follow (not the case)
* code is re-used (also not the case)

I think none of the above is applied in this case.